updated ClipboardItem and Clipboard documentation and examples using … (#33262)
mdn/content: files/en-us/web/api/clipboarditem/supports/index.md

---
title: "ClipboardItem: supports() method"
short-title: supports()
slug: Web/API/ClipboardItem/supports
page-type: web-api-static-method

> browser-compat: api.ClipboardItem.supports

Reviewer: I think this should be:

```suggestion
browser-compat: api.ClipboardItem.supports_static
```

to match the BCD data: https://github.com/mdn/browser-compat-data/blob/main/api/ClipboardItem.json#L173

---

{{APIRef("Clipboard API")}} {{securecontext_header}}

The **`supports()`** method of the {{domxref("ClipboardItem")}} interface returns true or false based on whether the specific {{Glossary("MIME type")}} type parameter is supprted by the clipboard.

## Syntax

```js-nolint
supports(type)
```

### Parameters

- `type`
  - : A valid {{Glossary("MIME type")}}.

### Return value

True or false value based on whether that {{Glossary("MIME type")}} is supported by the clipboard.

## Examples

> In the following example, we're writing data to the clipboard via the {{domxref("clipboard.write()")}} method. Then we utilize {{domxref("ClipboardItem.supports", "supports()")}} to see if the {{Glossary("MIME type")}} is supported by the clipboard before attempting to write it to the clipboard. Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unesseary CPU cycles from attempting to write unsupported formats to the clipboard. A `try..catch` block can be used to catch any errors while writing the data to the clipboard.

Reviewer 2: ```suggestion
### Writing an image to the clipboard

In the following example, we're writing an image to to the clipboard.

We use `supports()` to check whether the `"image/png"` MIME type is supported by the clipboard before fetching the image and writing it using {{domxref("clipboard.write()")}}.

We also use a [`try..catch`](/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) statement to catch any errors while writing the data to the clipboard.
```
 
- use a descriptive title for the example, and a sentence giving an overview of what we are doing
- don't self-link with `{{domxref("ClipboardItem.supports", "supports()")}}`
- avoid passive voice in the bit about `try...catch`, and link to the JS docs

I wasn't sure about "Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unnecessary CPU cycles from attempting to write unsupported formats to the clipboard." I think as a rule examples should just illustrate, that is they "show not tell" so things about the motivation for the method should live in the method general description. But I'm not sure this is needed there either, perhaps it is too obvious?

```js

> async function writeClipImg() {

Reviewer 2: ```suggestion
async function writeImageToClipboard() {
```

...this is strictly optional but I like more readable method names.

Reviewer 2: Oh I see the other examples use `writeClipImg` as well. So I guess this is even more optional :).

  try {
    if (ClipboardItem.supports('image/png')) {
        const imgURL = "/myimage.png";
        const data = await fetch(imgURL);
        const blob = await data.blob();
        await navigator.clipboard.write([
        new ClipboardItem({
            [blob.type]: blob,
        }),
        ]);
        console.log("Fetched image copied.");
    } else {
        console.log("image png is not suported");
    }
  } catch (err) {
    console.error(err.name, err.message);
  }
}
```

## Specifications

> {{Specifications}}

Reviewer 3: Spec link for `supports`: https://w3c.github.io/clipboard-apis/#dom-clipboarditem-supports

Reviewer: This link should come from BCD. The `{{Specifications}}` macro will pull it in automatically if it's present in BCD (browser-compat-data).

Reviewer: Just verified, the data is present: https://github.com/mdn/browser-compat-data/blob/main/api/ClipboardItem.json#L175C24-L175C88
